src/dawn/CodeGen/ASTCodeGenGTClangStencilDesc.cpp

```diff
@@ -62,12 +87,103 @@ void ASTCodeGenGTClangStencilDesc::visit(const std::shared_ptr<VerticalRegionDec
 
 void ASTCodeGenGTClangStencilDesc::visit(const std::shared_ptr<StencilCallDeclStmt>& stmt) {
   int StencilID = instantiation_->getStencilCallToStencilIDMap().find(stmt)->second;
+  StencilIDsVisited_.push_back(StencilID);
   for(const std::string& stencilName : StencilIDToStencilNameMap_.find(StencilID)->second)
     ss_ << std::string(indent_, ' ') << stencilName << ".get_stencil()->run();\n";
 }
 
 void ASTCodeGenGTClangStencilDesc::visit(const std::shared_ptr<BoundaryConditionDeclStmt>& stmt) {
-  DAWN_ASSERT_MSG(0, "BoundaryConditionDeclStmt not yet implemented");
+
+  auto calculateHaloExtents = [&](std::string fieldname) {
```

Comment:
General question, there is quite some algorithmic content here, that could belong to the compiler. 
Why dont we do all these in the optimizer, in the BC pass and store it in the BoundaryCondition struct of the StencilInstantiation?  Since this is important information that being stored in the ISIR can be used later on by anyone else, including other passes